src/mender-update/daemon/states.cpp

```diff
@@ -648,6 +648,15 @@ void UpdateBeforeCommitState::OnEnter(Context &ctx, sm::EventPoster<StateEvent>
 void UpdateCommitState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 	log::Debug("Entering ArtifactCommit state");
 
+	// Explicitly check if state scripts version is supported
+	auto rootfs_scripts = ctx.mender_context.GetConfig().paths.GetRootfsScriptsPath();
+	auto err = script_executor::CheckScriptsCompatibility(path::Join(rootfs_scripts, "version"));
```

Comment:
Minor, but I would consider letting the `CheckScriptsCompatibility` function add the `version` file, and only pass the path here, to have the logic be contained in the `scripts` package. But not overly important.

Reply:
It makes sense :+1: